src/string.cr

```diff
@@ -3377,12 +3377,13 @@ class String
   # Converts camelcase boundaries to underscores.
   #
   # ```
-  # "DoesWhatItSaysOnTheTin".underscore # => "does_what_it_says_on_the_tin"
-  # "PartyInTheUSA".underscore          # => "party_in_the_usa"
-  # "HTTP_CLIENT".underscore            # => "http_client"
-  # "3.14IsPi".underscore               # => "3.14_is_pi"
+  # "DoesWhatItSaysOnTheTin".underscore                         # => "does_what_it_says_on_the_tin"
+  # "PartyInTheUSA".underscore                                  # => "party_in_the_usa"
+  # "HTTP_CLIENT".underscore                                    # => "http_client"
+  # "3.14IsPi".underscore                                       # => "3.14_is_pi"
+  # "InterestingImage".underscore(Unicode::CaseOptions::Turkic) # => "ınteresting_ımage"
   # ```
-  def underscore
+  def underscore(options = Unicode::CaseOptions::None)
```

Comment:
You can add a type annotation

Reply:
I'd suggest to add it because it improves documentation. Readers can directly follow the type link to see available values.
And more importantly, it allows to use autocasted symbols for the enum members. This only works when the argument is typed as an enum type. So you can write `"InterestingImage".underscore(:turkic)`.

Reply:
Ooooh... I completely forgot about that awesome feature somehow! That's a good reason to add it.